### PR TITLE
Migrate image and file inputs to use Sanity UI

### PR DESCRIPTION
**Pull request by bjoerge**
Builds on #2346

### Description
​This migrates the file and image inputs to use Sanity UI under the hood.

Note: includes some chore commits at the beginning​

### What to review
Everything related to image and file fields​.
​
### Notes for release

- Migrates default image and file inputs in the studio to use [Sanity UI](https://sanity.io/ui) under the hood​

### CODE REVIEW

File: packages/@sanity/form-builder/src/inputs/common/fileTarget/utils/extractFiles.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
+/**
+ * Utilities for extracting files from dataTransfer in a predictable cross-browser fashion.
+ * Also recursively extracts files from a directory
+ * Inspired by https://github.com/component/normalized-upload
+ * Contains special handling of
```
Comment (rexxars) [inline packages/@sanity/form-builder/src/inputs/common/fileTarget/utils/extractFiles.ts:5]: of what

File: packages/@sanity/form-builder/src/inputs/common/fileTarget/utils/extractFiles.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+        // We disabled this for now, since we don't support handling datatransfer (e.g. pasting/drag+drop) of strings.
+        // todo: Consider adding support for this later on
+
+        // note: for security reasons reading `item.type` in a non-user initiated event stack returns an empty string
+        // so the next line is there to keep it in scope
+        // const mimeType = item.type
+        // return new Promise<string>((resolve) => item.getAsString(resolve)).then((str) => {
+        //   return str ? [new File([str], 'datatransfer.txt', {type: mimeType})] : []
+        // })
```
Comment (rexxars) [inline packages/@sanity/form-builder/src/inputs/common/fileTarget/utils/extractFiles.ts:65]: I propose we remove this - we have version control, after all? Maybe a small comment about what was there instead?

File: packages/@sanity/form-builder/src/inputs/files/ImageToolInput/ImageToolInput.tsx [hunk 3]
```diff
@@ -92,32 +89,19 @@ export default class ImageToolInput extends React.Component<
     return (
       // @todo: render presence and markers
       <ChangeIndicatorProvider
-        path={HOTSPOT_PATH}
-        focusPath={EMPTY_PATH}
+        path={['hotspot']}
+        focusPath={[]}
```
Comment (rexxars) [inline packages/@sanity/form-builder/src/inputs/files/ImageToolInput/ImageToolInput.tsx:93]: Do these have special handling now that doesn't cause re-rendering?

File: packages/@sanity/form-builder/src/inputs/files/common/FileInputButton/FileInputButton.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+          focusRing: theme.sanity.focusRing,
+        })};
+      }
+      &:focus:not(:focus-visible) {
+        box-shadow: ${focusRingBorderStyle(border)};
+      }
+    }
+
+    // This is a little hack to fix wonky @sanity/ui Button rendering where there's a child element.
+    // This issue should be fixed in @sanity/ui and this hack removed from here.
```
Comment (rexxars) [inline packages/@sanity/form-builder/src/inputs/files/common/FileInputButton/FileInputButton.tsx:53]: Did we create an issue for this under the Sanity UI repo or in the internal issue tracker?

**Review by rexxars — APPROVED** [review]
Looks good! A few minor comments but don't let them block a merge unless you agree.

File: packages/@sanity/form-builder/src/inputs/common/fileTarget/utils/extractFiles.ts (hunk 1 above)
Comment (bjoerge) [reply packages/@sanity/form-builder/src/inputs/common/fileTarget/utils/extractFiles.ts:5]: good question, I started this sentence three years ago and i have no idea to be honest 😅

File: packages/@sanity/form-builder/src/inputs/files/common/FileInputButton/FileInputButton.tsx (hunk 4 above)
Comment (bjoerge) [reply packages/@sanity/form-builder/src/inputs/files/common/FileInputButton/FileInputButton.tsx:53]: Good point! Created https://github.com/sanity-io/design/issues/471 and updated the comment to point at it.

File: packages/@sanity/form-builder/src/inputs/files/ImageToolInput/ImageToolInput.tsx (hunk 3 above)
Comment (bjoerge) [reply packages/@sanity/form-builder/src/inputs/files/ImageToolInput/ImageToolInput.tsx]: Good point, fixed to use never-changing constants.

**Pull request merged**